indexer/httptransport.go

```diff
@@ -116,8 +212,10 @@ func (h *HTTP) IndexHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	out := enc(w, r)
```

Comment:
@hdonnay 
Instead of calling this directly in a handler, can it become a middleware chain ?
If it becomes a middleware chain then we can also easily toggle Accept-Encoding handling via a flag and this struct's register's method.